src/v/cluster/metadata_cache.cc

```diff
@@ -217,6 +217,12 @@ std::optional<model::node_id> metadata_cache::get_controller_leader_id() {
 
 void metadata_cache::reset_leaders() { _leaders.local().reset(); }
 
+ss::future<> metadata_cache::refresh_health_monitor() {
+    co_await _health_monitor.invoke_on(
+      health_monitor_frontend::refresher_shard,
+      [](health_monitor_frontend& fe) { return fe.refresh_info(); });
+}
```

Comment:
i think that the health monitor should jump to the required core. the reason is that it reduces coupling between components. here the metadata cache has to know about the internal implementation of the health monitor frontend. if the monitor frontend handles that complexity for the caller, then the api is easier to use correctly.

Reply:
I can agree to that. I'll make the change.